examples/prometheus/prometheus.yaml

```diff
@@ -243,12 +252,22 @@ objects:
       - name: example-rules
         interval: 30s # defaults to global interval
         rules:
-        - alert: Node Down
+        - alert: NodeDown
           expr: up{job="kubernetes-nodes"} == 0
           annotations:
             miqTarget: "ContainerNode"
             severity: "HIGH"
-            message: "{{$labels.instance}} is down"
+            message: "{{$labels.instance}} is down"    
+        - alert: HangingBuild
+          expr: count(openshift_build_running_phase_start_time_seconds{} < time() - ${PROMETHEUS_ACTIVE_BUILD_ALERT_DURATION}) > ${PROMETHEUS_ACTIVE_BUILD_ALERT_THRESHOLD}
+          annotations:
+            severity: "MEDIUM"
+            message: "{{$labels.instance}} indicates at least ${PROMETHEUS_ACTIVE_BUILD_ALERT_THRESHOLD} OpenShift builds are taking more than ${PROMETHEUS_ACTIVE_BUILD_ALERT_DURATION} seconds to complete"
+        - alert: BuildFailureRate
+          expr: (count(openshift_build_terminal_phase_total{phase="failed"}) / count(openshift_build_terminal_phase_total{phase="complete"})) > ${PROMETHEUS_FAILED_BUILD_RATIO}
```

Comment:
How would this alert get cleared?  If openshift online has 9000 failed builds, do I have to wait for 9000 successful builds for the alert to go away?

Reply:
pending all the other discussion points in the PR, I almost did not include this alert to begin with (it was suggested to me) .... I'm inclined to simply remove it, but will wait a bit to see what other discussion arises with it

Reply:
As the suggester of the alert:  i don't know what the prometheus workflow is like, or how the alerting behaves... can you have an alert that's ignored once tripped until it resets below the alert level?

Pruning old builds would be one way for it to be reset.  Or we can punt it and do an alert based on "failed builds within the last X minutes" instead, but that's going to be another alert that needs to be tuned for your cluster/expectations.